crates/cargo-contract/src/cmd/info.rs

```diff
@@ -55,75 +62,90 @@ pub struct InfoCommand {
-                        (true, false) => println!("{}", info_to_json.to_json()?),
-                        (false, false) => {
-                            basic_display_format_contract_info(&info_to_json)
-                        }
-                        // Binary flag applied
-                        (_, true) => {
-                            let wasm_code =
-                                fetch_wasm_code(*info_to_json.code_hash(), &client)
-                                    .await?;
-                            match (wasm_code, self.output_json) {
-                                (Some(code), false) => {
-                                    std::io::stdout()
-                                        .write_all(&code)
-                                        .expect("Writing to stdout failed")
-                                }
-                                (Some(code), true) => {
-                                    let wasm = serde_json::json!({
-                                        "wasm": format!("0x{}", hex::encode(code))
-                                    });
-                                    println!(
-                                        "{}",
-                                        serde_json::to_string_pretty(&wasm).map_err(
-                                            |err| {
-                                                anyhow!(
-                                                    "JSON serialization failed: {}",
-                                                    err
-                                                )
-                                            }
-                                        )?
-                                    );
-                                }
-                                (None, _) => {
-                                    return Err(anyhow!(
-                                        "Contract wasm code was not found"
-                                    )
-                                    .into())
-                                }
-                            }
-                        }
+            // All flag applied
+            if self.all {
+                // Using count of 255 to avoid conversion issue to usize on
+                // any architecture
+                let count = 255;
+                let mut from = None;
+                let mut contracts = Vec::new();
+                loop {
+                    let len = contracts.len();
+                    contracts
+                        .append(&mut fetch_all_contracts(&client, count, from).await?);
+                    if contracts.len()
+                        < len
+                            + <u32 as TryInto<usize>>::try_into(count)
+                                .expect("Conversion from u32 to usize failed")
+                    {
+                        break
                     }
-                    Ok(())
+                    from = contracts.last();
```

Comment:
Will this return duplicate contracts over multiple loops? Have you tried it with a low `count`?

Reply:
Ok :+1: